diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/CommentsRule.kt

```diff
@@ -126,21 +126,19 @@ class CommentsRule(private val configRules: List<RulesConfig>) : Rule("comments"
      */
     private fun isCodeAfterCommentStart(text: String): Boolean {
         val textWithoutCommentStartToken = text.removePrefix("//").trim()
-        return textWithoutCommentStartToken.contains(classRegex) ||
-                textWithoutCommentStartToken.contains(importOrPackageRegex) ||
-                textWithoutCommentStartToken.contains(functionRegex) ||
-                textWithoutCommentStartToken.contains(rightBraceRegex)
+        return codeFileStartCases.any { textWithoutCommentStartToken.contains(it) }
     }
 
     companion object {
         private val importKeyword = KtTokens.IMPORT_KEYWORD.value
         private val packageKeyword = KtTokens.PACKAGE_KEYWORD.value
         private val importOrPackage = """($importKeyword|$packageKeyword) """.toRegex()
         private val classRegex =
-                """^\s*(public|private|open|internal|protected)*\s*(class|object)\s+(\w+)(\(.*\))*(\s*:\s*\w+(\(.*\))*)?\s*\{*$""".toRegex()
-        private val importOrPackageRegex = """^(import|package)?\s+([a-zA-Z.])+$""".toRegex()
-        private val functionRegex = """^(override)*\s?fun\s+\w+(\(.*\))?(\s*:\s*\w+)?\s*\{$""".toRegex()
+                """^\s*(public|private|open|internal|protected|data|sealed)*\s*(class|object)\s+(\w+)(\(.*\))*(\s*:\s*\w+(\(.*\))*)?\s*\{*$""".toRegex()
+        private val importOrPackageRegex = """^(import|package)?\s+([a-zA-Z.])+;*$""".toRegex()
+        private val functionRegex = """^(override|abstract|actual|expected)*\s?fun\s+\w+(\(.*\))?(\s*:\s*\w+)?\s*[{=]$""".toRegex()
```

Comment:
```suggestion
        private val functionRegex = """^(override|abstract|actual|expect)*\s?fun\s+\w+(\(.*\))?(\s*:\s*\w+)?\s*[{=]$""".toRegex()
```
It's actually [expect](https://kotlinlang.org/docs/reference/mpp-connect-to-apis.html). And expect/actual can be combined with visibility modifiers too.